java/arcs/core/data/SchemaFields.kt

```diff
@@ -65,4 +73,15 @@ val LARGEST_PRIMITIVE_TYPE_ID = PrimitiveType.values().size - 1
 data class SchemaFields(
     val singletons: Map<FieldName, FieldType>,
     val collections: Map<FieldName, FieldType>
-)
+) {
+
+    override fun toString() = toString(Type.ToStringOptions())
+
+    fun toString(options: Type.ToStringOptions) = "{${when (options.hideFields) {
+        true -> "..."
+        false -> listOf(
+            singletons.map { (name, type) -> "$name: $type" },
+            collections.map { (name, type) -> "$name: [$type]" }
+        ).flatten().joinToString()
+    }}}"
```

Comment:
Could be a bit more readable if the fields string were extracted to a variable.

```kotlin
    fun toString(options: Type.ToStringOptions): String {
        val fields = when (options.hideFields) {
            true -> "..."
            false -> listOf(
                singletons.map { (name, type) -> "$name: $type" },
                collections.map { (name, type) -> "$name: [$type]" }
            ).flatten().joinToString()
        }
        return "{$fields}"
    }
```

Reply:
Yup, I went too far with inlining :)